ObliviousHarmony/vscode-php-codesniffer
# Fix Running Executables With Spaces

### All Submissions:


### Changes proposed in this Pull Request:

Since `spawn` expects the first argument to be _only_ an executable, paths to PHPCS executables with spaces break it. This pull request fixes that bug by breaking the passed executable option into segments using spaces and quoted strings. The first segment is then used as the executable while the rest are prepended as arguments to pass before the PHPCS configuration arguments.

### How to test the changes in this Pull Request:

1. Disable `phpCodeSniffer.autoExecutable`.
2. Set `phpCodeSniffer.executable` to `php {path-to-phpcs}`, replacing `{path-to-phpcs}` with a path to a usable PHPCS executable in your workspace. For me this was `php vendor/bin/phpcs`.
3. Make sure that linting still works with the PR applied.
